.github/workflows/npm.yml

```diff
@@ -0,0 +1,263 @@
+    name: Test bindings on Linux-x64-gnu - node@${{ matrix.node }}
+    needs:
+      - build
+    strategy:
+      fail-fast: false
+      matrix:
+        node:
+          - '18'
+          - '20'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node }}
+          cache: yarn
+          cache-dependency-path: ./npm/napi/yarn.lock
+      - name: Install dependencies
+        run: yarn install
+        working-directory: ./npm/napi
+      - name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: bindings-x86_64-unknown-linux-gnu
+          path: ./npm/napi
+      - name: List packages
+        run: ls -R .
+        shell: bash
+        working-directory: ./npm/napi
+      - name: Test bindings
+        run: docker run --rm -v $(pwd):/build -w /build node:${{ matrix.node }}-slim yarn test
+        working-directory: ./npm/napi
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write # to publish provenance
+    needs:
+      - test-macOS-windows-binding
+      - test-linux-x64-gnu-binding
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
+          cache: yarn
+          cache-dependency-path: ./npm/napi/yarn.lock
+      - name: Install dependencies
+        run: yarn install
+        working-directory: ./npm/napi
+      - name: Download all artifacts
+        uses: actions/download-artifact@v3
+        with:
+          path: npm/napi/artifacts
+      - name: Move artifacts
+        run: yarn artifacts
+        working-directory: ./npm/napi
+      - name: List packages
+        run: ls -R ./npm
+        working-directory: ./npm/napi
+        shell: bash
+      - name: Publish Dry-Run
+        if: "!(github.repository == 'denoland/denokv' && startsWith(github.ref, 'refs/tags/'))"
+        run: |
+          npm config set provenance true
+          echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> ~/.npmrc
+          VERSION=0.0.0-dryrun DRY_RUN=1 ../.github/npm_publish.sh
+        working-directory: ./npm
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+      - name: Publish
+        if: github.repository == 'denoland/denokv' && startsWith(github.ref, 'refs/tags/')
+        run: |
+          echo "$TAG_NAME"
+          npm config set provenance true
+          if grep "^v\?[0-9]\+\.[0-9]\+\.[0-9]\+" <<< $TAG_NAME;
+          then
+            echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> ~/.npmrc
+            VERSION=$TAG_NAME ../.github/npm_publish.sh
+          else
+            echo "Not a release, skipping publish"
+          fi
+        working-directory: ./npm
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Good catch, was used by the default NAPI-RS template to do a GH release in prepublish.  Not needed here, since I already removed that part (you do GH releases in another workflow)

Verified not needed for npm provenance statements either, which _are_ still enabled in this workflow